A32NX/html_ui/Pages/A32NX_Utils/SimBriefApi.js

```diff
@@ -1,18 +1,36 @@
 class SimBriefApi {
-    static getSimBriefOfp(username, userId) {
-        let apiUrl;
-        if (username) {
-            apiUrl = `${SimBriefApi.url}&username=${username}`;
-        } else if (userId) {
-            apiUrl = `${SimBriefApi.url}&userid=${userId}`;
+    static getSimBriefOfp(userId) {
+        if (userId) {
+            return fetch(`${SimBriefApi.url}&userid=${userId}`)
+                .then((response) => {
+                    if (!response.ok) {
+                        throw new HttpError(response.status);
+                    }
+
+                    return response.json();
+                });
         } else {
-            throw ("No SimBrief username/user ID provided");
+            throw new Error("No SimBrief pilot ID provided");
+        }
+    }
+
+    static getSimBriefUser(value) {
+        if (!value) {
+            throw new Error("No SimBrief username/pilot ID provided");
+        }
+
+        // The SimBrief API will try both username and pilot ID if either one
+        // isn't valid, so request both if the input is plausibly a pilot ID.
+        let apiUrl = `${SimBriefApi.url}&username=${value}`;
+        if (/^\d{1,6}$/.test(value)) {
```

Comment:
What happens if SimBrief starts using 7-digit user ids? How about just using `\d+`?

Reply:
I'm happy to increase the number, but just allowing \d+ allows [stupidly long IDs](https://github.com/flybywiresim/a32nx/pull/2870#issuecomment-787461747) which don't make sense, and there's no way to discern between a user ID that exists but doesn't have an OFP available and one that doesn't exist at all in the API response.

If I increase this to 8 digits, is that a reasonable compromise?

Reply:
Discussed on Discord, increased to 8.